### ISSUE TITLE
Calls changing incorrectly defined

### ISSUE THREAD

**Issue by marcoscaceres** [sysapps/telephony#133]
The spec says:

>   Whenever a call is added to or removed from the <a>calls</a> array, the <a>user agent</a> MUST <a>queue a task</a> to <a>fire an event</a> named <code>callschanged</code>.

The above should be defined in terms of the underlying telephony system - not in terms of something changing in the calls array.

**Comment by zolkis** [comment]
Actually it is referring to the change in the 'calls' array only. New calls get signaled through 'onincoming', finished calls through 'ondisconnected'. 'calls' could be a synchronous method returning a sequence of TelephonyCall objects, at least my implementation does that.

**Comment by marcoscaceres** [comment]
I agree, we need to change call to a sequence. If `onincoming` handles an added call, and `ondisconnected` handles a call vanishing (I know, crappy word... bare with me), then `callschanged` seems redundant.

**Comment by zolkis** [comment]
What if the call disconnects, and you have it in the sequence (since it is copied by value), so it's invalid? I like more the idea of having exactly one call object per call, referenced.
